### DISCUSSION TITLE
Uncaught TypeError: "Snapshot missing on Livewire component with id: " is not a constructor in lazy component

### DISCUSSION THREAD

**Discussion by sajjadhossainshohag** [livewire/livewire#6235] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.0-beta.7

### Which PHP version are you using?

PHP 8.1

### Additional information

I am trying to show the product in the child component by loop, using lazy mode. The products show correctly with an error in the console and component action not working. But it works fine without lazy mode.
**My Code Screenshot Below:**
![code](https://github.com/livewire/livewire/assets/63788037/942bbdf2-1705-41d9-b04a-d8094555b4ee)

**Console screenshot below:**
![Screenshot 2023-08-07 094617](https://github.com/livewire/livewire/assets/63788037/503da816-76a2-4a07-8d12-9541ed3367d6)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by romansh** [comment, marked as answer]
Can you please provide a minimal working piece of code that reproduces your issue?

I just tried the next one, and it worked as expected.

```php
namespace App\Livewire;
use Livewire\Component;

class Test extends Component
{
    public array $loops = []; 
    public string $product = 'Product';

   public function mount()
   {
        $this->loops = [1,2,3];
   }

   public function render() { return <<<'HTML'
        <div>
        @forelse ($loops as $loop)
            <livewire:test2 :product="$product" lazy />
        @empty
            <div class="text-center">No products added to the wishlist.</div>
        @endforelse
    </div>
    HTML; }
}

class Test2 extends Component
{

    public $product;

    public function render() { return <<<'HTML'
        <div>{{ $product  }}</div>
    HTML; }
}
```

**Comment by devront-it** [comment]
Had the same issue, fixed it by adding wire:key to every foreach element and child component. So maybe this issue is related to dom diffing?

**Comment by ZebTheWizard** [comment]
Currently using `v3.0.0-beta.9` had a similar issue.  had to add `wire:key` to the looped element and the nested LW component like this

```
@foreach($posts as $post)
  <div class="border p-3" wire:key="{{ $post->id}}">
    <div>{{ $post->message }} </div>
    <livewire:like wire:key="{{ $post->id}}") />
  </div>
@endforeach
```
